packages/pipeline-editor/src/PipelineController/index.ts

```diff
@@ -592,7 +592,15 @@ class PipelineController extends CanvasController {
   updateProperties(nodeID: string, data: { [key: string]: any }) {
     const pipeline = this.findNodeParentPipeline(nodeID);
     if (pipeline !== undefined) {
-      const app_data = prefixedToNested(data);
+      const app_data: any = { component_parameters: {} };
+      const nested = prefixedToNested(data);
+      for (const [key, val] of Object.entries(
+        nested.component_parameters ?? {}
+      )) {
+        if (val !== undefined && val !== null) {
+          app_data.component_parameters[key] = val;
```

Comment:
does this still handle the case where we set the value back to undefined (I forget which properties we do that one)?

Reply:
I refactored this a little due to a test failure, I don't think that would happen with updated code but you could take a look in case